### DISCUSSION TITLE
Would Moose work well as a distributed storage for SQLite?

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#526] [Q&A]
mounted on nodes of docker swarm

**Comment by tianon** [comment]
You'll need to be careful that you're not expecting MooseFS to provide distributed _write_ access to SQLite (because filesystem-based locking is going to be a challenge), but for either a single service or read-only access it should work fine.

(I run Plex on top of my MooseFS instance, and it uses SQLite pretty extensively without issues.)

**Reply by eleaner** [reply]
I am thinking to spin seafile. There should be a single instance of it running - but swarm is not perfect

**Reply by thebigbone** [reply]
How do you handle multiple docker containers (in swarm) doing rw operations to the db then?

**Comment by thebigbone** [comment]
Sorry for bumping a 2 year old discussion but @eleaner did you test out moosefs on docker swarm? I currently mounted the storage directories on each node and when using sqlite for an application, it corrupts the db.

**Reply by tianon** [reply]
"There should be a single instance of it running" https://github.com/moosefs/moosefs/discussions/526#discussioncomment-5160788

(sqlite isn't designed for multi-writer - if you need that, sqlite isn't the right solution)

**Reply by thebigbone** [reply]
Yes, sorry I should have mentioned that I was trying out postgres. Not sqlite. It works fine if I start one replica of postgres on one node and then spawn multiple replicas of applications on multiple nodes doing rw operations on the db.

**Comment by ptempier** [comment]
For cases like this i think it s better to use mfsbdev.
You "should" get more perfs.
It will restrain you to use the block device on a single host at a time,
but you can't open sqlite with multiple clients anyway.

**Reply by thebigbone** [reply]
I didn't know about that. Thanks, will have a look at it.

**Reply by ptempier** [reply]
also for case you need lot of small files, like 100k small files.
but you only need them from a  single host at a time.
you trade that for like 10x speed.
